biogeochem/EDPhysiologyMod.F90

```diff
@@ -1245,92 +1254,322 @@ subroutine phenology_leafonoff(currentSite)
-      enddo !currentCohort
+          currentCohort => currentCohort%shorter
+       enddo !currentCohort
 
-      currentPatch => currentPatch%younger
+       currentPatch => currentPatch%younger
 
-   enddo !currentPatch
+    enddo !currentPatch
 
   end subroutine phenology_leafonoff
 
+  ! =====================================================================================
+
+  subroutine satellite_phenology(currentSite, bc_in)
+
+    ! -----------------------------------------------------------------------------------
+    ! Takes the daily inputs of leaf area index, stem area index and canopy height and 
+    ! translates them into a FATES structure with one patch and one cohort per PFT
+    ! The leaf area of the cohort is modified each day to match that asserted by the HLM 
+    ! -----------------------------------------------------------------------------------
+
+    ! !USES:                                                                                                       
+    !                                                                                                              
+    ! !ARGUMENTS:                                                                                                  
+    type(ed_site_type), intent(inout), target :: currentSite
+    type(bc_in_type),   intent(in)            :: bc_in
+
+    class(prt_vartypes), pointer :: prt
+
+    ! !LOCAL VARIABLES:                                                                                            
+    type(ed_patch_type) , pointer :: currentPatch     
+    type(ed_cohort_type), pointer :: currentCohort
+
+    real(r8) ::  spread        ! dummy value of canopy spread to estimate c_area
+    real(r8) ::  leaf_c        ! leaf carbon estimated to generate target tlai
+    real(r8) :: check_treelai
+    integer ::   fates_pft     ! fates pft numer for weighting loop
+    integer  ::   hlm_pft      ! host land model pft number for weighting loop.
+    integer ::   s             ! site index
+
+
+    ! To Do in this routine.
+    ! Get access to HLM input varialbes. 
+    ! Weight them by PFT
+    ! Loop around patches, and for each single cohort in each patch 
+    ! call assign_cohort_SP_properties to determine cohort height, dbh, 'n',  area, leafc from drivers.
+
+    currentSite%sp_tlai(:) = 0._r8
+    currentSite%sp_tsai(:) = 0._r8
+    currentSite%sp_htop(:) = 0._r8
+
+    ! WEIGHTING OF FATES PFTs on to HLM_PFTs
+    ! 1. Add up the area associated with each FATES PFT 
+    ! where pft_areafrac is the area of land in each HLM PFT and (from surface dataset)
+    ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
+
+    ! 2. weight each fates PFT target for lai, sai and htop by the area of the 
+    ! contrbuting HLM PFTs.
+
+    currentPatch => currentSite%oldest_patch
+    do while (associated(currentPatch))
+
+       fates_pft = currentPatch%nocomp_pft_label
+       if(fates_pft.ne.0)then 
```

Comment:
This is impossible in sp right? I see a check on it below.  Can we remove this clause then?